robot/Cumulus/tests/browser/donations/create_donation_payment.robot

```diff
@@ -1,36 +1,68 @@
-    Click Related List Button    Payments    New
+    [Documentation]  Navigate to Opportunities page and open an Opportunity>In the right sections, go to Payments click on drop down Arrow>New
+    ...              Create a new payment for the Opportunity.
+
+    [tags]                                          W-038461                 feature:Donations
+
+    Go To Page                                      Listing
+    ...                                             Opportunity
+
+    Click Object Button                             New
+    Select Record Type                              Donation
+
+    Create Opportunities                            Sravani $100 donation
+    ...                                             ${data}[contact][LastName] Household
+    ...                                             Closed Won
+
+    Save Current Record ID For Deletion             Opportunity
+    Current Page Should Be                          Detail                                  Opportunity
+
+    Verify Payments Made                            0
+
+    #Make A New Payment
+
+    Click Related List Button                       Payments                                New
+    Wait For Modal                                  New                                     Payment
     Select Window
-    Populate Field    Payment Amount    100
-    Select Value From Dropdown   Payment Method              Credit Card
-    Open Date Picker    Payment Date
-    Pick Date    Today
-    Click Modal Button        Save
-    Verify Occurrence    Payments    1
+    Populate Modal Form                             Payment Amount=100
+    ...                                             Payment Method=Credit Card
+
+    Open Date Picker                                Payment Date
+    Pick Date                                       Today
+    Click Modal Button                              Save
+
+    Verify Payments Made                            1
+
+    Go To Page                                      Details
+    ...                                             Contact
+    ...                                             object_id=${data}[contact][Id]
+
+    #Perform Validations
     ${opp_date} =     Get Current Date    result_format=%-m/%-d/%Y
-    Go To Record Home  &{contact}[Id]
-    Scroll Element Into View    text:Donation Totals
-    Confirm Field Value           Last Gift Date    contains    ${opp_date}    
-    Scroll Element Into View    text:Soft Credit Total
-    Confirm Field Value          Total Gifts    contains    $100.00    
-    Confirm Field Value           Total Number of Gifts    contains    1    
\ No newline at end of file
+    Validate Field Value Under Section   Donation Totals      Last Gift Date                  ${opp_date}
+    Validate Field Value Under Section   Soft Credit Total    Total Gifts                     $100.00
+    Validate Field Value Under Section   Soft Credit Total    Total Number of Gifts           1
```

Comment:
This name is misleading as Total Gifts and Total number of gifts are under Donation Totals not soft credit total. we are scrolling little under the field names to make the fields visible. instead u could scroll to that particular element and then confirm value. you can add that step of scrolling to the field before confirming the value in the confirm field value keyword itself or make this little more meaningful.

Reply:
Changed to a meaningful name